pkg/controller/apimanager/apimanager_controller_test.go

```diff
@@ -0,0 +1,110 @@
+package apimanager
+
+import (
+	"context"
+	"testing"
+
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/operator"
+	appsv1alpha1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
+	appsv1 "github.com/openshift/api/apps/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestAPIManagerControllerCreate(t *testing.T) {
+	var (
+		name           = "example-apimanager"
+		namespace      = "operator-unittest"
+		wildcardDomain = "test.3scale.net"
+	)
+
+	apimanager := &appsv1alpha1.APIManager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: appsv1alpha1.APIManagerSpec{
+			APIManagerCommonSpec: appsv1alpha1.APIManagerCommonSpec{
+				WildcardDomain: wildcardDomain,
+			},
+		},
+	}
+
+	// Objects to track in the fake client.
+	objs := []runtime.Object{apimanager}
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(appsv1alpha1.SchemeGroupVersion, apimanager)
+	err := appsv1.AddToScheme(s)
+	if err != nil {
+		t.Fatalf("Unable to add Apps scheme: (%v)", err)
+	}
+	imagev1.AddToScheme(s)
+	if err != nil {
+		t.Fatalf("Unable to add Image scheme: (%v)", err)
+	}
+	routev1.AddToScheme(s)
+	if err != nil {
+		t.Fatalf("Unable to add Route scheme: (%v)", err)
+	}
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewFakeClient(objs...)
```

Comment:
The fake client basically instantly records the object as created when it finds a Create request. This means the update of the "status" section will always succeed(there will not be any requeue in that case)